### DISCUSSION TITLE
Reset fakes guidance

### DISCUSSION THREAD

**Discussion by felpel** [FakeItEasy/FakeItEasy#1991] [Q&A] [answered]
Hi 👋 We've recently switched from `Moq` to `FakeItEasy` and it has mostly been smooth and easy to migrate to.

However, I have had difficulties concerning **reset of fakes**, especially with what is being suggested [here](https://fakeiteasy.github.io/docs/8.1.0/advanced-usage/#clearing-the-configuration-of-a-fake).

> discard the fake and create a new one

If I have a private member in a fixture class, does it mean to do something like such?

```csharp
// This can be called in `Dispose` context and co
private void ResetFakes()
{
      this.MyFakedDependency = A.Fake<IFakedDependency>(/*Our customization*/);
}
```

The alternative might have been to clear the recorded calls with `Fake.ClearConfiguration` as suggested [here](https://fakeiteasy.github.io/docs/8.1.0/advanced-usage/#clearing-a-fakes-recorded-calls), but I doubt I should be leveraging this given the `Obsolete` attribute message:
![image](https://github.com/FakeItEasy/FakeItEasy/assets/5000004/95beeccf-08f2-4379-a57a-d1e25e8239b6)

**Comment by thomaslevesque** [comment, marked as answer]
Hi @felpel,

Glad to count a new FakeItEasy user!

> If I have a private member in a fixture class, does it mean to do something like such?
> ```csharp
> // This can be called in `Dispose` context and co
> private void ResetFakes()
> {
>       this.MyFakedDependency = A.Fake<IFakedDependency>(/*Our customization*/);
> }

Yes, that's right. Note that if you also created other objects using the old fake, you will need to recreate them as well.

> The alternative might have been to clear the recorded calls with Fake.ClearConfiguration as suggested [here](https://fakeiteasy.github.io/docs/8.1.0/advanced-usage/#clearing-a-fakes-recorded-calls), but I doubt I should be leveraging this given the Obsolete attribute message:

If you just want to clear the recorded calls, use `Fake.ClearRecordedCalls`, which isn't obsolete.

If you want to reset the fake configuration, you could use `Fake.ClearConfiguration`, but as you noticed, it's planned for removal (in fact we intended to remove it in 8.0.0, and didn't... oops). The reason is that it only kinda works, and making it "really work" isn't easy; I won't go into the details, there are several issues where it's discussed. We have plans to offer a new way to reset fakes, but it hasn't seen much progress lately...

**Reply by felpel** [reply]
> If you just want to clear the recorded calls, use `Fake.ClearRecordedCalls`, which isn't obsolete.

What happens when you read the docs too fast ... 🙄 this did the trick :tada:
